### ISSUE TITLE
default <non-nullable-T> value for nullable T params should report fatal error

### ISSUE THREAD

**Issue by liesauer** [peachpiecompiler/peachpie#574]
```php
<?php

function test(?string $param = 0) {
    var_dump($param);
}
```

expect at compile-time
```
PHP Fatal error:  Default value for parameters with a string type can only be string or NULL
```

actually:
no error and no warning

but i haven't tested what's going to happen at runtime.

**Comment by jakubmisek** [comment]
thanks @liesauer
